pkg/ffcapi/api.go

```diff
@@ -40,6 +41,9 @@ type API interface {
 	// NextNonceForSigner is used when there are no outstanding transactions for a given signing identity, to determine the next nonce to use for submission of a transaction
 	NextNonceForSigner(ctx context.Context, req *NextNonceForSignerRequest) (*NextNonceForSignerResponse, ErrorReason, error)
 
+	// GasEstimate provides an estimate of the gas required for the given transaction
+	GasEstimate(ctx context.Context, req *ethsigner.Transaction) (*GasEstimateResponse, ErrorReason, error)
```

Comment:
@matthew1001 - just wondering why this isn't exactly the same pattern as:

https://github.com/hyperledger/firefly-transaction-manager/blob/0e611061a7d2daeeaeb0c821fb1584c457b4ccd2/pkg/ffcapi/transaction_prepare.go#L41-L43

Reply:
Yeah you're right, it should be. Updated in latest commit.